File: leetcode/cpp/2894-divisibleAndNon-divisibleSumsDifference.cpp
#include <iostream>
using namespace std;

class Solution {
    public:
        int differenceOfSums(int n, int m) {
            int res = 0;
            for (int i = 0; i <= n; i++) {
                if (i % m) {
                    res += i;
                }
                else {
                    res -= i;
                }
            }

            return res;
        }
    };

int main() {
    Solution sol;
    int n = 10, m = 3;
    cout << sol.differenceOfSums(10, 3);
    return 0;
}

File: leetcode/python/2894-divisibleAndNon-divisibleSumsDifference.py
class Solution:
    def differenceOfSums(self, n: int, m: int) -> int:
        res = sum([i * (1 if i % m else -1) for i in range(n + 1)])
        return res
    
def main():
    sol = Solution()
    n, m = 10, 3
    print(sol.differenceOfSums(n, m))

if __name__ == "__main__":
    main()